### ISSUE TITLE
JS Plugin Context Fails When Wrapped by External Plugins

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#15325]
When using JS plugins that wrap the `context` object (like `eslint-plugin-mozilla`), accessing `context.sourceCode` from the wrapped context throws a `TypeError: Cannot read private member #internal from an object whose class did not declare it`.

This breaks compatibility with ESLint plugins that create wrapper rules by copying or proxying the context object.

## Reproduction

The issue occurs when a plugin rule wraps another rule and passes a modified context. Here's a minimal reproduction:

```typescript
// Plugin that wraps another rule (simulates eslint-plugin-mozilla pattern)
const baseRule = {
  create(context) {
    const sourceCode = context.sourceCode; // This will fail
    return { /* visitor */ };
  },
};

const wrappingRule = {
  create(context) {
    // Create wrapped context by copying properties
    const wrappedContext = Object.create(Object.getPrototypeOf(context));
    for (const key of Object.keys(context)) {
      wrappedContext[key] = context[key];
    }
    
    // Pass wrapped context to base rule
    return baseRule.create(wrappedContext); // Error occurs here
  },
};
```

**Config:**
```json
{
  "jsPlugins": ["eslint-plugin-mozilla"],
  "rules": {
    "mozilla/no-redeclare-with-import-autofix": "error"
  }
}
```

## Error Output

```
x Error running JS plugin.
| File path: /path/to/file.js
| TypeError: Cannot read private member #internal from an object whose class did not declare it
|     at getInternal (file:///path/to/oxlint/dist/plugins.js:9242:29)
|     at get sourceCode (file:///path/to/oxlint/dist/plugins.js:9193:3)
```

## Root Cause

The `Context` class in `apps/oxlint/src-js/plugins/context.ts` uses a private field `#internal` to store internal data:

```typescript
export class Context {
  #internal: InternalContext;
  
  get sourceCode(): SourceCode {
    getInternal(this, 'access `context.sourceCode`');
    return SOURCE_CODE;
  }
}
```

The `getInternal` function accesses `context.#internal`:

```typescript
getInternal = (context, actionDescription) => {
  const internal = context.#internal; // Fails if context is not original instance
  if (internal.ruleIndex === -1) throw new Error(...);
  return internal;
};
```

When a plugin wraps the context object (even using `Object.create(Object.getPrototypeOf(context))`), the private `#internal` field is not preserved. Private class fields are not enumerable and cannot be copied through property enumeration or prototype chains.

This is a common pattern in ESLint plugins:
- **eslint-plugin-mozilla**: Rules like `no-redeclare-with-import-autofix` wrap ESLint's core `no-redeclare` rule
- Other plugins may proxy or extend contexts to override methods like `report()`

## Affected Plugins

- `eslint-plugin-mozilla` (confirmed)
- Potentially any plugin that wraps or extends context objects

**Comment by overlookmotel** [comment]
Fixed in #15448.

Thanks for getting to the bottom of the cause and for the test cases @camc314.
